src/server/services/input-types.js

```diff
@@ -115,11 +125,21 @@ types.Date = input => {
 };
 
 types.Array = (input, params) => {
-    const [innerType] = params;
-    if (!Array.isArray(input)) throw GENERIC_ERROR;
+    const [_innerType, min, max] = params;
+    const innerType = grabType(_innerType);
```

Comment:
Maybe renaming this to
```javascript
const [typeParam, min, max] = params;
const innerType = getTypeParser(typeParam);
```
Thoughts?

Reply:
If you add some default values for `min`, `max`, you can probably simplify some of the code further:
```javascript
types.Array = (input, params=[]) => {
    const [typeParam, min=0, max=Infinity] = params;
    const innerType = getTypeParser(typeParam);
```